lib/builderator/config.rb

```diff
@@ -0,0 +1,81 @@
+require_relative './config/file'
+require_relative './config/defaults'
+
+module Builderator
+  ##
+  # Global Configuration
+  ##
+  module Config
+    class << self
+      ## GLOBAL_DEFAULTS is the lowest-precedence layer, followed by dynamicly
+      ## defined instance-defaults.
+      def layers
+        @layers ||= []
+      end
+
+      def defaults
+        @defaults ||= File.new({}, :source => 'dafaults', :config => self)
+      end
+
+      def overrides
+        @overrides ||= File.new({}, :source => 'overrides', :config => self)
+      end
+
+      def argv(options = {})
+        @argv ||= File.new(options, :source => 'overrides', :config => self)
+      end
+
+      def append(path)
+        layers << File.from_file(path, :config => self) if ::File.exist?(path)
+      end
+      alias_method :load, :append
+
+      def append_json(path)
```

Comment:
I'm not a fan of encoding type information in method names, since it leads to an explosion of methods as more supported types are added. I want to be able to write code like

```ruby
Config.load 'config.rb'
Config.load 'config.json'
Config.load 'config.ini'
```

and have the `Config::File` class make a best guess as to what type of file it's loading. If it makes the wrong guess, I should be able to explicitly specify type.

```ruby
Config.load 'config.properties', :type => :java
```